### ISSUE TITLE
Geweke and numba problems

### ISSUE THREAD

**Issue by ColCarroll** [arviz-devs/arviz#726]
**Describe the bug**
It looks like this is two problems:
1. Calling `az.geweke` throws a bunch of numba warnings
2. Even if numba is disabled, the warnings get thrown, which makes me think it is not respecting the numba flag

**To Reproduce**
```python
import arviz as az
import numpy as np

print(az.Numba.numba_flag)   # True

az.Numba.disable_numba()
print(az.Numba.numba_flag)  # False

az.geweke(np.random.randn(100))  # emits a big warning from numba
```

Here's the warning:
```
/home/petethepup/projects/arviz/arviz/stats/diagnostics.py:457: NumbaWarning: 
Compilation is falling back to object mode WITH looplifting enabled because Function "geweke" failed type inference due to: Untyped global name '_geweke': cannot determine Numba type of <class 'function'>

File "../../projects/arviz/arviz/stats/diagnostics.py", line 498:
def geweke(ary, first=0.1, last=0.5, intervals=20):
    <source elided>
    # Filter out invalid intervals
    return _geweke(ary, first, last, intervals)
    ^

  @conditional_jit
/home/pete/miniconda3/envs/arviz3.6/lib/python3.6/site-packages/numba/compiler.py:725: NumbaWarning: Function "geweke" was compiled in object mode without forceobj=True.

File "../../projects/arviz/arviz/stats/diagnostics.py", line 458:
@conditional_jit
def geweke(ary, first=0.1, last=0.5, intervals=20):
^

  self.func_ir.loc))
/home/petethepup/miniconda3/envs/arviz3.6/lib/python3.6/site-packages/numba/compiler.py:734: NumbaDeprecationWarning: 
Fall-back from the nopython compilation path to the object mode compilation path has been detected, this is deprecated behaviour.

For more information visit http://numba.pydata.org/numba-doc/latest/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit

File "../../projects/arviz/arviz/stats/diagnostics.py", line 458:
@conditional_jit
def geweke(ary, first=0.1, last=0.5, intervals=20):
^
```

**Expected behavior**
I think *with* numba enabled, no warning should be emitted (that might be my installation, and a numba problem). Without numba enabled, I think there should be no warning at all.

**Additional context**
This is running on master, numba `0.44.1`

**Comment by ColCarroll** [comment]
(tagging @Ban-zee, who might have thoughts on this)

**Comment by ahartikainen** [comment]
What numba version do you have?

**Comment by ColCarroll** [comment]
0.44.1

**Comment by ahartikainen** [comment]
I had 0.42 and there were no warnings. After update, I see many warnings :)

**Comment by ahartikainen** [comment]
Ok, at least warnings are fixable, just some typing wrapping apparently

https://numba.pydata.org/numba-doc/dev/proposals/type-inference.html

Now the leakage... let's see

**Comment by ahartikainen** [comment]
```
def conditional_jit(function=None, **kwargs):  # noqa: D202
    """Use numba's jit decorator if numba is installed.
    Notes
    -----
        If called without arguments  then return wrapped function.
        @conditional_jit
        def my_func():
            return
        else called with arguments
        @conditional_jit(nopython=True)
        def my_func():
            return
    """

    def wrapper(function):
        try:
            numba = importlib.import_module("numba")
            return numba.jit(**kwargs)(function)

        except ImportError:
            return function

    if function:
        return wrapper(function)
    else:
        return wrapper
```

Conditional jit should have numba check before wrapper is called? Or called? Now it returns numba function if it's installed.

**Comment by Ban-zee** [comment]
Working on it.

**Comment by ColCarroll** [comment]
Fixed by #728
